### PR TITLE
JAMES-1895: JMAP and MailboxSession type

### PR DESCRIPTION
**Pull request by quynhn**
- JMAP should use userSession instead of systemSession
- Add more testing on messageIdManager for system session

### CODE REVIEW

**Comment by quynhn** [comment]
test this please

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java [hunk 1]
```diff
@@ -372,6 +372,11 @@ public MailboxSession createSystemSession(String userName, Logger log) {
         return createSession(userName, null, log, SessionType.System);
     }
 
+    @Override
+    public MailboxSession createUserSession(String userName, Logger log) throws BadCredentialsException {
+        return createSession(userName, null, log, SessionType.User);
```
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java]: What is this _null_ parameter?

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java [hunk 2]
```diff
@@ -272,6 +272,22 @@
...
+     *
+     * @param userName
+     *            the name of the user whose session is being created
+     * @param log
+     *            context sensitive log
+     * @return <code>MailboxSession</code>, not null
+     * @throws BadCredentialsException
+     *             when system access is not allowed for the given user
+     */
+    MailboxSession createUserSession(String userName, Logger log) throws BadCredentialsException;
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java]: createUserSessionBypassAuthentication ?

I would like to make it clear...

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 3]
```diff
@@ -243,7 +251,7 @@ public void searchShouldNotReturnResultsFromOtherNamespaces() throws Exception {
 
     @ContractTest
     public void searchShouldNotReturnResultsFromOtherUsers() throws Exception {
-        session = mailboxManager.createSystemSession(USER_1, LoggerFactory.getLogger("Mock"));
+        session = mailboxManager.createSystemSession(USER_1, LOGGER);
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java]: Have it as a field ?

Or unnecessary changes...

**Comment by quynhn** [comment]
This PR is invalid because:
```
for the moment a system session is nothing more than an internal user session (ie a user session without a login process)
what the point of having a real system session with advanced rights?
I think this will ask several hard questions, such as quota handling, etc.
```
